PilotDataPlatform/cli
# Patch20240802: Fixup listing related logic

## Summary

after testing with dataset/project listing command, the pagination and numbering are not correctly formatted. The PR includes following updates:
 - Raise error when target dataset version is not available.
 - Add total number of projects in listing command.
 - Add total number of datasets in listing command.

## JIRA Issues

[PILOT-5697](https://indocconsortium.atlassian.net/browse/PILOT-5697)

## Type of Change

Please delete options that are not relevant.


## Testing

Are there any new or updated tests to validate the changes?


## Test Directions

update output check in old test cases


[PILOT-5697]: https://indocconsortium.atlassian.net/browse/PILOT-5697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
